test/metabase/driver/h2_test.clj

```diff
@@ -159,31 +160,36 @@
                       "ORDER BY ATTEMPTS.DATE ASC")
                  (some-> (qp/compile query) :query pretty-sql))))))))
 
-(deftest classify-ddl-test
+(deftest check-single-select-statement-test
   (mt/test-driver :h2
-    (are [query] (= false (#'h2/contains-ddl? (u/the-id (mt/db)) query))
-      "select 1"
-      "update venues set name = 'bill'"
-      "delete venues"
-      "select 1;
-       update venues set name = 'bill';
-       delete venues;")
-
-    (is (= nil (#'h2/check-disallow-ddl-commands
-                {:database (u/the-id (mt/db))
-                 :engine :h2
-                 :native {:query (str/join "; "
-                                           ["select 1"
-                                            "update venues set name = 'bill'"
-                                            "delete venues"])}})))
-    (let [trigger-creation-attempt
-          (str/join "\n" ["DROP TRIGGER IF EXISTS MY_SPECIAL_TRIG;"
-                          "CREATE OR REPLACE TRIGGER MY_SPECIAL_TRIG BEFORE SELECT ON INFORMATION_SCHEMA.Users AS '';"
-                          "SELECT * FROM INFORMATION_SCHEMA.Users;"])]
-      (is (thrown?
-           IllegalArgumentException
-           #"DDL commands are not allowed to be used with h2."
-           (#'h2/check-disallow-ddl-commands
+    (testing "a single select statement should pass"
+      (is (nil?
+           (#'h2/check-single-select-statement
             {:database (u/the-id (mt/db))
              :engine :h2
-             :native {:query trigger-creation-attempt}}))))))
+             :native {:query "select 1"}}))))
+    (testing "not a single select statement"
+      (doseq [query ["update venues set name = 'bill'"
+                     "insert into venues (name) values ('bill')"
+                     "delete venues"
+                     "select 1; update venues set name = 'bill'; delete venues;"
+                     (str/join "\n" ["DROP TRIGGER IF EXISTS MY_SPECIAL_TRIG;"
+                                     "CREATE OR REPLACE TRIGGER MY_SPECIAL_TRIG BEFORE SELECT ON INFORMATION_SCHEMA.Users AS '';"
+                                     "SELECT * FROM INFORMATION_SCHEMA.Users;"])]]
+        (is (thrown?
+             IllegalArgumentException
+             #"Only a single SELECT statement is allowed."
+             (#'h2/check-single-select-statement
+              {:database (u/the-id (mt/db))
+               :engine :h2
+               :native {:query query}})))))))
+
+(deftest check-admin-commands-test
```

Comment:
This isn't related to the `metabase.driver.h2` namespace, but I wanted to confirm and document that these commands don't run on H2 databases without admin rights